### PR TITLE
Batch export as JSON support

### PR DESCRIPTION
**Pull request by TeaCrab**
I personally wanted the app to have the capability of simply batch extract as JSON.  So I added the function in at the cost of existing command-line functions.  Command-line functions now only handle batch export as JSON.

The edits to EqCrt.cs and EqCus.cs are to stop the app from throwing errors when trying to export those 2 file types as JSON.  Regardless of which file, Armor or Weapon, the editor function tries to access non-existent dictionary keys in all/both contexts which throws an error because keys can't be found.

### CODE REVIEW

File: Windows/MainWindow.xaml.cs [hunk 1]
```diff
@@ -47,6 +48,8 @@ public partial class MainWindow {
 
         [CanBeNull] private CancellationTokenSource savedTimer;
 
+        public static string file_name = "none";
```
Comment (Synthlight) [inline Windows/MainWindow.xaml.cs:51]: The already exists as `targetFile`. If you need only the filename from that, `Path.GetFileName(targetFile)` or `Path.GetFileNameWithoutExtension(targetFile)`.

File: Windows/MainWindow.xaml.cs [hunk 2]
```diff
@@ -95,11 +98,31 @@ public partial class MainWindow {
             }
         }
 
-        public bool SingleClickToEditMode { get; set; } = true;
```
Comment (Synthlight) [inline Windows/MainWindow.xaml.cs:98]: Don't change the defaults.

File: Windows/MainWindow.xaml.cs [hunk 3]
```diff
@@ -223,6 +247,74 @@ public partial class MainWindow {
                 panel.UpdateLayout();
             }
         }
+        private void ExportJson(string oFile = null)
+        {
```
Comment (Synthlight) [inline Windows/MainWindow.xaml.cs:251]: Please put this on the same line as everything else. I don't want mixed styles in here.

File: Windows/MainWindow.xaml.cs [hunk 4]
```diff
@@ -256,7 +348,7 @@ public partial class MainWindow {
 
             var loadData = targetFileType.GetMethod("LoadData", BindingFlags.Static | BindingFlags.Public | BindingFlags.FlattenHierarchy);
             Debug.Assert(loadData != null, nameof(loadData) + " != null");
-            fileData = (IMhwMultiStructFile) loadData.Invoke(null, new object[] {targetFile});
+            fileData = (IMhwMultiStructFile)loadData.Invoke(null, new object[] {targetFile});
```
Comment (Synthlight) [inline Windows/MainWindow.xaml.cs:351]: Please don't make pointless changes to my style like this.

File: Windows/MainWindow.xaml.cs [hunk 5]
```diff
@@ -472,6 +564,18 @@ public partial class MainWindow {
             }
         }
 
+        [CanBeNull]
```
Comment (Synthlight) [inline Windows/MainWindow.xaml.cs:567]: IS there some reason this couldn't stay exactly where it was already?

File: Windows/MainWindow.xaml.cs [hunk 6]
```diff
@@ -538,21 +642,11 @@ public partial class MainWindow {
...
-            }
-            return $"^{string.Join(@"\|", split)}$";
-        }
-
         private async void SaveJson(bool mergeWithTarget) {
             if (string.IsNullOrEmpty(targetFile)) return;
 
             try {
-                var         fileName      = Path.GetFileName(targetFile);
+                var fileName = Path.GetFileName(targetFile);
```
Comment (Synthlight) [inline Windows/MainWindow.xaml.cs:649]: Please don't make pointless changes to my style like this.

File: Windows/MainWindow.xaml.cs [hunk 7]
```diff
@@ -95,11 +98,31 @@ public partial class MainWindow {
...
+                    string iPath = Path.GetDirectoryName(arg);
+                    string iFileName = Path.GetFileName(arg).Replace(".", "_");
+                    string oPath = Path.Combine(iPath, "_Json");
+                    string oFileName = iFileName + ".json";
+                    System.IO.Directory.CreateDirectory(oPath);
+                    string target = Path.Combine(oPath, oFileName);
+                    BatchLoad(arg);
+                    ExportJson(target);
+                }
+                System.Environment.Exit(0);
```
Comment (Synthlight) [inline Windows/MainWindow.xaml.cs:122]: This doesn't need to be exclusive. You could, for example, add a cmd flag for batch exporting and try to load every arg after it.

File: Windows/MainWindow.xaml.cs [hunk 8]
```diff
@@ -149,7 +173,7 @@ public partial class MainWindow {
 
             UpdateCheck.Run(this);
 
-            TryLoad(args);
+            // TryLoad(args);
```
Comment (Synthlight) [inline Windows/MainWindow.xaml.cs:176]: Check for some batch export cmd flag instead, that way you don't disable this feature.

File: Structs/Items/EqCus.cs [hunk 9]
```diff
@@ -16,8 +16,8 @@ public partial class Entries : IHasArmorType, IHasWeaponType {
             [DisplayName("Equipment Category")]
             [IsReadOnly]
             public ArmorType Equipment_Category_Armor {
-                get => (ArmorType) Equipment_Category_Raw;
-                set => Equipment_Category_Raw = (byte) value;
+                get => (ArmorType)Equipment_Category_Raw;
```
Comment (Synthlight) [inline Structs/Items/EqCus.cs:19]: More unnecessary changes.

File: Windows/MainWindow.xaml.cs (hunk 1 above)
Comment (TeaCrab) [reply Windows/MainWindow.xaml.cs:51]: Sorry, I don't remember seeing this change on the diff of the pull request...

File: Windows/MainWindow.xaml.cs (hunk 2 above)
Comment (TeaCrab) [reply Windows/MainWindow.xaml.cs:98]: I'm only trying to honor your work by contributing my modifications to it.  I understand that you don't have to accept every little change I have modified.

File: Windows/MainWindow.xaml.cs (hunk 4 above)
Comment (TeaCrab) [reply Windows/MainWindow.xaml.cs:351]: Sorry again, I'm not aware of this being a change...  Please ignore it and have it the way you like.

File: Windows/MainWindow.xaml.cs (hunk 5 above)
Comment (TeaCrab) [reply Windows/MainWindow.xaml.cs:567]: No, this must be me failing to rebase properly or something.  Sorry again.

File: Windows/MainWindow.xaml.cs (hunk 6 above)
Comment (TeaCrab) [reply Windows/MainWindow.xaml.cs:649]: Again, I'm sorry if I made this change.  I agree that it is absolutely pointless.

File: Windows/MainWindow.xaml.cs (hunk 3 above)
Comment (TeaCrab) [reply Windows/MainWindow.xaml.cs:251]: Not sure what happened here...

File: Windows/MainWindow.xaml.cs (hunk 8 above)
Comment (TeaCrab) [reply Windows/MainWindow.xaml.cs:176]: I wasn't sure how this worked, sorry.

File: Structs/Items/EqCus.cs (hunk 9 above)
Comment (TeaCrab) [reply Structs/Items/EqCus.cs:19]: Absolutely.

File: Windows/MainWindow.xaml.cs (hunk 7 above)
Comment (TeaCrab) [reply Windows/MainWindow.xaml.cs:122]: Ahhh too much work.

**Comment by Synthlight** [comment]
Sorry I let this sit in limbo for so long.

Basically, I don't want to add another thing called "json export" since users get confused enough by the already existing json export.
It also interferes with other cmd line params.

Thanks for your work, but I won't be including this.

**Comment by TeaCrab** [comment]
Ehh, don't worry about it.  I just like to write codes from time to time.
By the time I finish, I pretty much lost interest in what I actually wanted
to do.  lol


Brent Liu
Link for portfolio <https://www.artstation.com/artist/remotecrab131>
